assets/js/components/surveys/SurveyQuestionOpenText.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * SurveyQuestionOpenText component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+import { useInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import SurveyHeader from './SurveyHeader';
+import { TextField, Input, HelperText } from '../../material-components';
+import Button from '../Button';
+import VisuallyHidden from '../VisuallyHidden';
+
+const MAXIMUM_CHARACTER_LIMIT = 100;
+
+const SurveyQuestionOpenText = ( {
+	question,
+	answerQuestion,
+	placeholder,
+	subtitle,
+	dismissSurvey,
+	submitButtonText,
+} ) => {
+	const [ value, setValue ] = useState( '' );
+
+	const handleSubmit = () => {
+		answerQuestion( { answer: value } );
+	};
+
+	const onChange = useCallback(
+		( event ) => {
+			setValue( event.target.value?.slice( 0, MAXIMUM_CHARACTER_LIMIT ) );
```

Comment:
This doesn't need optional chaining though, right? I think you mentioned in another PR this wasn't needed, can we remove it here too?
```suggestion
			setValue( event.target.value.slice( 0, MAXIMUM_CHARACTER_LIMIT ) );
```

If it's actually needed, then we should do `|| ''` or this could be `undefined`.

Reply:
Removed and the tests pass. :+1: